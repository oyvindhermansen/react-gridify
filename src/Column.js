import React, { Component } from 'react'
import { css } from 'glamor'

export default class Column extends Component {
  calculateWidth(size) {
    return 100 / size
  }

  largify(gridSize) {
    const styles = css({
      width: `${this.calculateWidth(gridSize)}%`,
      float: 'left'
    })

    return styles
  }

  mediumify(gridSize) {
    const styles = css({
      '@media(max-width: 900px)': {
        width: `${this.calculateWidth(gridSize)}%`,
        float: 'left'
      }
    })

    return styles
  }

  smallify(gridSize) {
    const styles = css({
      '@media(max-width: 600px)': {
        width: `${this.calculateWidth(gridSize)}%`,
        float: 'left'
      }
    })

    return styles
  }

  paddify(padding) {
    let p = css({
      boxSizing: 'border-box',
      paddingLeft: '5px',
      paddingRight: '5px'
    })
    if (padding) {
      p = css({
        boxSizing: 'border-box',
        paddingLeft: padding,
        paddingRight: padding
      })
    }
    return p
  }

  render() {
    const {
      small,
      medium,
      large,
      padding,
      className,
      children
    } = this.props

    const additionalClassNames = className ? className : ''

    return (
      <div
        className={`column ${additionalClassNames}`}
        {...this.largify(large)}
        {...this.mediumify(medium)}
        {...this.smallify(small)}
        {...this.paddify(padding)}
      >
        {children}
      </div>
    )
  }
}
