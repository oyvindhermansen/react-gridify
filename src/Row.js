import React, { Component } from 'react';
import { css } from 'glamor';

export default class Row extends Component {
  setStyles(maxWidth) {
    let styles = {
      maxWidth: '64rem',
      width: '100%',
      margin: '0 auto',
      display: 'table'
    }

    if (maxWidth) {
      styles = Object.assign(
        styles,
        { maxWidth: maxWidth }
      )
    }

    return css(styles)
  }

  render() {
    const { children, maxWidth } = this.props;
    return (
      <div className="row" {...this.setStyles(maxWidth)}>
        {children}
      </div>
    )
  }
}
