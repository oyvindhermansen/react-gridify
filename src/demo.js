import React, {Component} from 'react';
import { render } from 'react-dom';
import { Row, Column } from '../lib/index';

class App extends Component {

	constructor(props) {
		super(props)
		this.state = {
			items: [
				{ id: 1, text: 'Hello World 1' },
				{ id: 2, text: 'Hello World 2' },
				{ id: 3, text: 'Hello World 3' },
				{ id: 4, text: 'Hello World 4' },
				{ id: 5, text: 'Hello World 5' },
				{ id: 6, text: 'Hello World 6' },
				{ id: 7, text: 'Hello World 7' },
				{ id: 8, text: 'Hello World 8' },
				{ id: 9, text: 'Hello World 9' },
				{ id: 10, text: 'Hello World 10' },
				{ id: 11, text: 'Hello World 11' },
				{ id: 12, text: 'Hello World 12' },
			]
		}
	}

	render() {
		const styles = {
			padding: '1rem',
			margin: '0.5rem',
			backgroundColor: 'orange'
		}
		return (
			<div>
				<Row>
					{
						this.state.items.map((item, index) => {
							return (
								<Column key={item.id} small="2" medium="6" large="12">
									<div style={styles}>{item.text}</div>
								</Column>
							)
						})
					}
				</Row>
			</div>
		);
	}
}

document.addEventListener('DOMContentLoaded', () => {
  render(
    <App />,
    document.getElementById('root')
  );
});
