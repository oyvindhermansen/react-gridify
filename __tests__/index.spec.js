import React from 'react';
import renderer from 'react-test-renderer';
import { Row, Column } from '../src/index';

describe('React gridify', () => {

  describe('Row', () => {
    it('should match snapshot', () => {
      const component = renderer.create(
        <Row>This is children</Row>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Column', () => {
    it('should match snapshot', () => {
      const component = renderer.create(
        <Column
          centered
          padding="10px"
          xSmall="12"
          small="6"
          medium="4"
          large="3">
          This is children
        </Column>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })

    it('adding additional classNames properly', () => {
      const component = renderer.create(
        <Column
          className="myAdditionalClass"
          large="12">
        </Column>
      );
      let tree = component.toJSON();

      expect(tree.props.className).toBe('column myAdditionalClass');
    });

    it('should throw when adding column numbers < 12', () => {
      function componentFailure() {
        const component = renderer.create(
          <Column large="15"></Column>
        );
        return component;
      }

      expect(() => {
        componentFailure()
      }).toThrow()
    });
  });

});
