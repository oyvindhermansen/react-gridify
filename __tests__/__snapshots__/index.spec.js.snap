// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React gridify Column should match snapshot 1`] = `
<div
  className="column"
  data-css-1lwsn8m=""
  data-css-1p8eac0=""
  data-css-1wgphrl=""
  data-css-7s6x1j=""
  data-css-b9k42t=""
>
  This is children
</div>
`;

exports[`React gridify Row should match snapshot 1`] = `
<div
  className="row"
  data-css-g7t3c5=""
>
  This is children
</div>
`;
